tests/components/group/test_notify.py

```diff
@@ -1,58 +1,105 @@
+    hass: HomeAssistant,
+    tmp_path: Path,
+    async_get_service: Any = None,
+):
+    """Specialize the mock platform for legacy notify service."""
+    loaded_platform = MockNotifyPlatform(async_get_service)
+    mock_platform(hass, "test.notify", loaded_platform)
+
+    return loaded_platform
+
+
+async def help_setup_notify(
+    hass: HomeAssistant,
+    tmp_path: Path,
+    targets: dict[str, None] | None = None,
+    group_setup: list[dict[str, None]] | None = None,
+) -> MagicMock:
+    """Help set up a platform notify service."""
+    send_message_mock = MagicMock()
+
+    class _TestNotifyService(notify.BaseNotificationService):
+        def __init__(self, targets: dict[str, None] | None) -> None:
+            """Initialize service."""
+            self._targets = targets
+            super().__init__()
+
+        @property
+        def targets(self) -> Mapping[str, Any] | None:
+            """Return a dictionary of registered targets."""
+            return self._targets
+
+        def send_message(self, message: str, **kwargs: Any) -> None:
+            """Send a message."""
+            send_message_mock(message, kwargs)
+
+    async def async_get_service(
+        hass: HomeAssistant,
+        config: ConfigType,
+        discovery_info: DiscoveryInfoType | None = None,
+    ) -> notify.BaseNotificationService:
+        """Get notify service for mocked platform."""
+        return _TestNotifyService(targets)
+
+    # Mock platform with service
+    mock_notify_platform(hass, tmp_path, async_get_service=async_get_service)
+    # Setup the platform
+    items: list[dict[str, Any]] = [{"platform": "test"}]
+    items.extend(group_setup or [])
+    await async_setup_component(hass, "notify", {"notify": items})
+    await hass.async_block_till_done()
 
-    service1.send_message = MagicMock(autospec=True)
-    service2.send_message = MagicMock(autospec=True)
+    # Return mock for assertion service calls
+    return send_message_mock
 
-    def mock_get_service(hass, config, discovery_info=None):
-        if config["name"] == "demo1":
-            return service1
-        return service2
 
+async def test_send_message_with_data(hass: HomeAssistant, tmp_path: Path) -> None:
+    """Test sending a message with to a notify group."""
+    send_message_mock = await help_setup_notify(
+        hass, tmp_path, {"service1": 1, "service2": 2}
+    )
     assert await async_setup_component(
         hass,
         "group",
         {},
     )
     await hass.async_block_till_done()
 
-    with patch.object(demo, "get_service", mock_get_service):
-        await async_setup_component(
-            hass,
-            notify.DOMAIN,
-            {
-                "notify": [
-                    {"name": "demo1", "platform": "demo"},
-                    {"name": "demo2", "platform": "demo"},
-                ]
-            },
-        )
-        await hass.async_block_till_done()
-
     service = await group.async_get_service(
```

Comment:
Side note: We can probably remove this call since we setup the platform already above and then below call the `notify.test_service1` etc instead.

Reply:
I am not sure as this this the `group` call. The only thing I changed is setting up mocked notify services instead of using the `demo` instances.

Reply:
I mean that we should include the group platform in the config dict when we set up the notify integration with `async_setup_component`.

https://www.home-assistant.io/integrations/group/#notify-groups

Reply:
Improved version: https://github.com/home-assistant/core/pull/115526